hrm-domain/lambdas/custom-integrations/uscr-dispatcher/index.ts

```diff
@@ -14,17 +14,115 @@
  * along with this program.  If not, see https://www.gnu.org/licenses/.
  */
 
-// eslint-disable-next-line prettier/prettier
 import type { ALBEvent, ALBResult } from 'aws-lambda';
+import { isErr } from '@tech-matters/types';
+import { validateEnvironment, validateHeaders, validatePayload } from './validation';
+import { authenticateRequest } from './authentication';
+import { logger } from './logger';
+import * as hrmService from './hrm-service';
+import * as beaconService from './beacon-service';
+import * as mapping from './mapping';
 
 export const handler = async (event: ALBEvent): Promise<ALBResult> => {
   try {
-    console.debug(`custom-integrations/uscr-dispatcher: called with event ${event}`);
+    console.log(JSON.stringify(event));
+    const envResult = validateEnvironment();
+    if (isErr(envResult)) {
+      const message = `${envResult.error} ${envResult.message}`;
+      logger({ message, severity: 'error' });
+      return { statusCode: 500, body: message };
+    }
+
+    const payloadResult = validatePayload(JSON.parse(event.body || '{}'));
+    if (isErr(payloadResult)) {
+      const message = `${payloadResult.error} ${payloadResult.message}`;
+      logger({ message, severity: 'warn' });
+      return { statusCode: 400, body: message };
+    }
+
+    const headersResult = validateHeaders(event.headers);
+    if (isErr(headersResult)) {
+      const message = `${headersResult.error} ${headersResult.message}`;
+      logger({ message, severity: 'warn' });
+      return { statusCode: 400 };
+    }
+
+    const authResult = await authenticateRequest({
+      accountSid: payloadResult.data.accountSid,
+      authHeader: headersResult.data.authToken,
+      environment: envResult.data.environment,
+    });
+    if (isErr(authResult)) {
+      const message = authResult.error + authResult.message;
+      logger({ message, severity: 'warn' });
+      return { statusCode: 401 };
+    }
+
+    // Get (or create) case associated with the contact so we can track the incident being reported
+    const createCaseResult = await hrmService.getOrCreateCase({
+      accountSid: payloadResult.data.accountSid,
+      casePayload: payloadResult.data.casePayload,
+      contactId: payloadResult.data.contactId,
+      token: authResult.data.token,
+    });
+    if (isErr(createCaseResult)) {
+      const message = createCaseResult.error + createCaseResult.message;
+      logger({ message, severity: 'error' });
+      return { statusCode: 500 };
+    }
+
+    const { contact, caseObj } = createCaseResult.data;
+
+    logger({ message: JSON.stringify(createCaseResult), severity: 'info' });
+
+    // Case already contains a corresponding case entry section, we asume the incident has been created but something went wrong updating HRM. Poller will eventually bring consitency to this case
+    if (hrmService.hasIncidentCaseSection(caseObj)) {
```

Comment:
I'm slightly wary that this check doesn't cover all the edge cases - that there could be scenarios where an incident exists on case but one still needs adding for this dispatch. But it's probably ok for now if beacon think it is, those scenarios have probably been descoped, just want to be sure